File: Day13/PerfectSquareChecker.java
package Day13;
import java.util.Scanner;

public class PerfectSquareChecker {
    public static boolean getisPerfectNumber(int num){
        if(num <0 ){
            return false; // Negative number can not be a perfect square
        }
        int i =0;
        while(i*i <= num){
        if(i*i == num){
            return true;
        }
        i++;
    }
    return false;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number: ");
        int num = sc.nextInt();
        sc.close();
        boolean ans = getisPerfectNumber(num);
        if(ans)
            System.out.println(num + " is a perfect square");
        else
            System.out.println(num + " is not a perfect square");
    }
}

File: Day13/perfect_square_checker.c
#include<stdio.h>

// Function to check if a number is a perfect square
int isPerfectSquare(int num){
    if(num <0 ){
        return 0; // Negative number can not be a perfect square
    }
    int i =0;
    while(i*i <= num){
        if(i*i == num){
            return 1;
        }
        i++;
    }
    return 0;
}
int main(){
    int num;
    printf("Enter the number: ");
    scanf("%d",&num);
    //Store the result in a variable and display it
    int ans = isPerfectSquare(num);
    if(ans){
        printf("%d is a perfect square.",num);
    }
    else{
        printf("%d is not a perfect square.",num);
    }
    return 0;
}
